src/xcode/ENA/ENA/Source/Views/BaseElements/ENAButton.swift

```diff
@@ -108,6 +114,52 @@ class ENAButton: DynamicTypeButton {
 	private func applyHighlight() {
 		highlightView.isHidden = !isHighlighted
 	}
+
+	private func applyActivityIndicator() {
+		guard isLoading else {
+			activityIndicator?.removeFromSuperview()
+			return
+		}
+
+		guard nil == activityIndicator else { return }
+
+		let activityIndicator = UIActivityIndicatorView(style: .medium)
+		activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+		activityIndicator.isUserInteractionEnabled = false
+
+		addSubview(activityIndicator)
+
+		if let title = titleLabel {
+			title.leadingAnchor.constraint(equalTo: activityIndicator.trailingAnchor, constant: 8).isActive = true
+			activityIndicator.leadingAnchor.constraint(greaterThanOrEqualTo: self.leadingAnchor, constant: 8).isActive = true
+			activityIndicator.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+		} else {
+			activityIndicator.centerXAnchor.constraint(equalTo: centerXAnchor).isActive = true
+			activityIndicator.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+			activityIndicator.leadingAnchor.constraint(greaterThanOrEqualTo: self.leadingAnchor, constant: 8).isActive = true
+			self.trailingAnchor.constraint(greaterThanOrEqualTo: activityIndicator.trailingAnchor, constant: 8).isActive = true
```

Comment:
Are these two constraints really needed if we have ```centerXAnchor``` and ```centerXAnchor``` already?

Reply:
Yes, otherwise the button may collapse in width.